src/main/java/sysc4806/graduateAdmissions/model/Privilege.java

```diff
@@ -0,0 +1,37 @@
+package sysc4806.graduateAdmissions.model;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+/**
+ * This class is used to specify privileges for roles. A Role with have a collection
+ * of these objects to determine what it can and cannot do. A Privilege is made up of three
+ * parts: An action (eg. create, delete), a target for that action (eg. application, interest),
+ * and an owner of that target (eg. self, everyone). Combinations of these allow the
+ * specification of many different privileges for roles.
+ *
+ * For example, a student would have privileges for creating and deleting their own applications.
+ * Admin staff would have privileges for creating and deleting any student's application.
+ *
+ * @author luke
+ */
+@Data
+@Builder
+public class Privilege {
```

Comment:
Just also want to note that it's totally cool to have @ NoArgsContructor and @ AllArgsConstructor annotations as well as @ Data and @ Builder all on the same class. I'm sorry, I see I made you change the tests to accommodate this but I didn't mean to : S

No change required

Reply:
No worries. I've never used lombok, and I just took the first thing it provided that looked like it could do what I wanted. I didn't even see the builder annotation when I first wrote this. I think the build annotation a lot more because you can create objects with any number of parameters filled (instead of writing a bunch of setters after creating the object).